### PR TITLE
Remove timeout from query param

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Currently timeout is passed, which is useless
`?allowWatchBookmarks=true&resourceVersion=1&timeout=8m26s&timeoutSeconds=506&watch=true'`

```release-note
none
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/rest/request.go [hunk 1]
```diff
@@ -468,7 +468,8 @@ func (r *Request) URL() *url.URL {
 	}
 
 	// timeout is handled specially here.
-	if r.timeout != 0 {
+	// Do not set timeout for GET request as apiserver does not use it.
+	if r.timeout != 0 && r.verb != "GET" {
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/rest/request.go]: this is not correct, the line in the API server is for watch only

**Comment by liggitt** [comment]
thanks for the PR, but the change to client-go is not correct. it stops sending a timeout parameter for all GET requests which would be honored by the server for everything but watch

/close

**Comment by hzxuzhonghu** [comment]
> for all GET requests which would be honored by the server for everything but watch

I looked into the apiserver get handler and can not find any of that kind of use. Can you show where it is used?

**Comment by liggitt** [comment]
Hmm, you might be right. however, special-casing GET requests in client code doesn't really make sense. It's better to send the configured timeout consistently so the server knows the client's expected timeout. We could use that information in the future, and if we drop it from client requests now, it will be almost impossible to consistently update all clients in the future to start sending it again.

**Comment by hzxuzhonghu** [comment]
I just happened to see this duplicate param `timeout` with `timeoutSeconds`.

**Pull request closed without merging**